### ISSUE TITLE
Support multi ModelCheckpoint checkpoint_on metrics  

### ISSUE THREAD

**Issue by tchaton** [PyTorchLightning/pytorch-lightning#3879]
## 🚀 Feature

Currently, callback ModelCheckpoint allows to save the best model based on one provided metric to monitor.
The new feature proposal would be to extend this to support multiple monitored metrics to checkpoint on.

### Motivation

When a torch-lightning weights is used for paper benchmarking, it would be possible to make an inference with the best model for a given metric.

### Pitch

Approach 1:

```
checkpoint_callback_1 = ModelCheckpoint(monitor='val_loss',
        ...     filepath='my/path/sample-mnist-{epoch:02d}-{val_loss:.2f}'
        ... )

checkpoint_callback_2 = ModelCheckpoint(monitor='val_miou,
        ...     filepath='my/path/sample-mnist-{epoch:02d}-{val_miou:.2f}'
        ... )

checkpoint_callback_n = ModelCheckpoint(monitor='val_acc,
        ...     filepath='my/path/sample-mnist-{epoch:02d}-{val_acc:.2f}'
        ... )

 trainer = Trainer(callbacks=[checkpoint_callback_1, ... checkpoint_callback_n])
```

or 

Approach 2 (prefered):

Extend monitor, save_top_k, mode, period to support list inputs like

```
checkpoint_multi_monitor_callback = ModelCheckpoint(monitor=['val_loss', val_miou],
                                             save_top_k =[1, 2], # or could be broadcasted if int provided
                                              mode=["min", "max"] # or could be broadcasted if str provided
                                              period=[1, 2],  # or could be broadcasted if int provided
        ...     filepath='my/path/sample-mnist-{epoch:02d}-{monitor:.2f}'
        ... )

 trainer = Trainer(callbacks=[checkpoint_multi_monitor_callback])
```

Also,

```
ModelIO.load_from_checkpoint
```

could be updated to support loading the weights for a specified monitored metrics.

```
    @classmethod
    def load_from_checkpoint(
        cls,
        checkpoint_path: str,
        *args,
        map_location: Optional[Union[Dict[str, str], str, torch.device, int, Callable]] = None,
        hparams_file: Optional[str] = None,
        strict: bool = True,
        monitored_metric: str = None, # If provided, it will try to load from the best weight from this metric.
        **kwargs,
    ):
```

**Comment by tchaton** [comment]
Hey @williamFalcon, any thoughts on it ?

**Comment by ananthsub** [comment]
This should come after the 1.0 release. For option 1, the trainer needs to ensure that the checkpoint callbacks are the last callbacks to run. For option 2, properties like the best model path/score need to be extended.

**Comment by rohitgr7** [comment]
IMO Approach 1 seems easy to work with, plus Approach 2 will force to have the same custom directory right?

**Comment by tchaton** [comment]
Hey @rohitgr7, you are totally right ! Approach 1 would enable to have minimum API changes and work with custom directory for monitored metric.
